### PR TITLE
CI | NSFS | Ceph S3 Tests | Move tests from `nsfs_s3_tests_pending_list` to `nsfs_s3_tests_black_list`

### PR DESCRIPTION
**Pull request by shirady**
### Explain the changes
1. Moved tests of features that are not supported in NSFS from `nsfs_s3_tests_pending_list` (tests of supported features that we want to investigate in the future) to `nsfs_s3_tests_black_list` (tests of unsupported features that fail and we don't need to investigate) - those tests include the following names:
- `post_object`
- `test_lifecycle` (except `test_lifecycle_get_no_id` and `test_lifecycle_set` see #8341 )
- `object_lock_`
- `test_sse`
- `test_cors`

### Issues: 
1. None, we want to have a pending list with the tests that we need to investigate, currently we also had failed tests of features that are not supported in NSFS.
2. GAP - we still need to do an additional scan the pending list as there might be more tests of unsupported features in NSFS.

### Testing Instructions:
1. It's running in the CI, on a local machine can use: `make test-nsfs-cephs3 CONTAINER_PLATFORM=linux/arm64` (I'm using the flag `CONTAINER_PLATFORM=linux/arm64` because I run it on my Mac M1).

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_black_list.txt [hunk 1]
```diff
@@ -349,4 +349,63 @@ s3tests_boto3/functional/test_sts.py::test_assume_role_with_web_identity_resourc
...
+s3tests_boto3/functional/test_s3.py::test_post_object_missing_policy_condition
+s3tests_boto3/functional/test_s3.py::test_post_object_request_missing_policy_specified_field
+s3tests_boto3/functional/test_s3.py::test_post_object_expired_policy
+s3tests_boto3/functional/test_s3.py::test_post_object_invalid_request_field_value
+s3tests_boto3/functional/test_s3.py::test_post_object_authenticated_request_bad_access_key
+s3tests_boto3/functional/test_s3.py::test_sse_s3_default_post_object_authenticated_request
+s3tests_boto3/functional/test_s3.py::test_sse_kms_default_post_object_authenticated_request
+s3tests_boto3/functional/test_s3.py::test_post_object_upload_size_rgw_chunk_size_bug
+s3tests_boto3/functional/test_s3.py::test_post_object_wrong_bucket
+s3tests_boto3/functional/test_s3.py::test_lifecycle_get_no_id
```
Comment (romayalon) [inline src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_black_list.txt:364]: lifecycle set and get should not already work?

Comment (shirady) [reply src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_black_list.txt:364]: @romayalon If you think that there are tests that should not be moved, please mention them explicitly.

1) I ran the test `test_lifecycle_get_no_id` and it failed on `An error occurred (InvalidArgument) when calling the PutBucketLifecycleConfiguration operation: Invalid Argument` 

From the logs:

> Sep-17 12:08:09.749 [nsfs/1539] [ERROR] core.endpoint.s3.ops.s3_put_bucket_lifecycle:: Rule should not have prefix, it should be filter.prefix { Expiration: [ { Days: [ '31' ] } ], Prefix: [ 'test1/' ], Status: [ 'Enabled' ] }
> Sep-17 12:08:09.750 [nsfs/1539] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InvalidArgument</Code><Message>Invalid Argument</Message><Resource>/shira-1r9zmzz8r6wpwizpkpmprc4-1?lifecycle</Resource><RequestId>m167ohvy-cmbso9-12xz</RequestId></Error> PUT /shira-1r9zmzz8r6wpwizpkpmprc4-1?lifecycle {"host":"localhost:6443","accept-encoding":"identity","user-agent":"Boto3/1.24.96 Python/3.11.9 Darwin/23.4.0 Botocore/1.27.96","content-md5":"jrj/iU+jjpkrfmC3IBOyIw==","x-amz-date":"20240917T090809Z","x-amz-content-sha256":"eae0e8d85d076012d6229e9dd31fcab1c54c6da18f03a1a982f53d3ae5610a13","authorization":"AWS4-HMAC-SHA256 Credential=<>/20240917/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=58b90beb6b996adc378daf65b2520d6520edd6d43efcb072b415de79145b8f85","amz-sdk-invocation-id":"865f1205-3a22-407a-a2b5-d3fb131f96dc","amz-sdk-request":"attempt=1","content-length":"298"} Error: Invalid Argument
>     at /Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/ops/s3_put_bucket_lifecycle.js:99:19
>     at arrayMap (/Users/shiradymnik/SourceCode/noobaa-core/node_modules/lodash/lodash.js:653:23)
>     at Function.map (/Users/shiradymnik/SourceCode/noobaa-core/node_modules/lodash/lodash.js:9622:14)
>     at Object.put_bucket_lifecycle [as handler] (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/ops/s3_put_bucket_lifecycle.js:79:31)
>     at handle_request (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/s3_rest.js:161:28)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async Object.s3_rest [as handler] (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/s3_rest.js:66:9

2) I ran the test `Invalid Argument` and it also failed on ` An error occurred (InvalidArgument) when calling the PutBucketLifecycleConfiguration operation: Invalid Argument`

From the logs:

> Sep-17 12:13:32.822 [nsfs/3517] [ERROR] core.endpoint.s3.ops.s3_put_bucket_lifecycle:: Rule should not have prefix, it should be filter.prefix { ID: [ 'rule1' ], Expiration: [ { Days: [ '1' ] } ], Prefix: [ 'test1/' ], Status: [ 'Enabled' ] }
> Sep-17 12:13:32.823 [nsfs/3517] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InvalidArgument</Code><Message>Invalid Argument</Message><Resource>/shira-7ntvyrwfxkkgdb5f569c5kw-1?lifecycle</Resource><RequestId>m167vf67-dlum3h-dcr</RequestId></Error> PUT /shira-7ntvyrwfxkkgdb5f569c5kw-1?lifecycle {"host":"localhost:6443","accept-encoding":"identity","user-agent":"Boto3/1.24.96 Python/3.11.9 Darwin/23.4.0 Botocore/1.27.96","content-md5":"ODH6PufzOpp/mWS8cdW01Q==","x-amz-date":"20240917T091332Z","x-amz-content-sha256":"88c7f2b2714656ee77a13fe6e3a72dd4793b75a81efb2ca6e39791c3d2dbc5fa","authorization":"AWS4-HMAC-SHA256 Credential=<>/20240917/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=b3bf5108417016fc8b21efbb7ddbc52376c86d210e19d4c44db02f90c08f50d0","amz-sdk-invocation-id":"e2cc265a-a55d-4179-83f7-c73bd3ea4389","amz-sdk-request":"attempt=1","content-length":"324"} Error: Invalid Argument
>     at /Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/ops/s3_put_bucket_lifecycle.js:99:19
>     at arrayMap (/Users/shiradymnik/SourceCode/noobaa-core/node_modules/lodash/lodash.js:653:23)
>     at Function.map (/Users/shiradymnik/SourceCode/noobaa-core/node_modules/lodash/lodash.js:9622:14)
>     at Object.put_bucket_lifecycle [as handler] (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/ops/s3_put_bucket_lifecycle.js:79:31)
>     at handle_request (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/s3_rest.js:161:28)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async Object.s3_rest [as handler] (/Users/shiradymnik/SourceCode/noobaa-core/src/endpoint/s3/s3_rest.js:66:9)

In containerized both tests are in the pending list.
Do you want to leave them in the pending list of NSFS as well?

Comment (romayalon) [reply src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_black_list.txt:364]: @shirady see - https://github.com/noobaa/noobaa-core/issues/8341, so we will support it soon - so pending list indeed, i'll add these tests addition suggestion on the bug.

Comment (shirady) [reply src/test/system_tests/ceph_s3_tests/s3-tests-lists/nsfs_s3_tests_black_list.txt]: Moved back, thanks.

**Review by romayalon — APPROVED** [review]

**Pull request merged**